File: backend/models/jobs.py
from typing import Any, Dict, List, Optional

from pydantic import BaseModel, Field


class JobApplyOption(BaseModel):
  title: Optional[str]
  link: Optional[str]


class JobHighlight(BaseModel):
  title: Optional[str]
  items: List[str] = Field(default_factory=list)


class JobListing(BaseModel):
  job_id: Optional[str] = None
  htidocid: Optional[str] = None
  title: Optional[str] = None
  company: Optional[str] = None
  location: Optional[str] = None
  via: Optional[str] = None
  description: Optional[str] = None
  posted_at: Optional[str] = None
  salary: Optional[str] = None
  extensions: List[str] = Field(default_factory=list)
  detected_extensions: Dict[str, Any] = Field(default_factory=dict)
  apply_options: List[JobApplyOption] = Field(default_factory=list)
  job_highlights: List[JobHighlight] = Field(default_factory=list)
  share_link: Optional[str] = None


class JobSearchResponse(BaseModel):
  query: str
  location: str
  page: int
  employment_type: Optional[str] = None
  role_filters: List[str] = Field(default_factory=list)
  jobs: List[JobListing]


class JobDetailResponse(BaseModel):
  job: JobListing

File: frontend/src/types/jobs.ts
export type JobApplyOption = {
  title?: string | null;
  link?: string | null;
};

export type JobHighlight = {
  title?: string | null;
  items: string[];
};

export type JobListing = {
  job_id?: string | null;
  htidocid?: string | null;
  title?: string | null;
  company?: string | null;
  location?: string | null;
  type?: string | null;
  via?: string | null;
  description?: string | null;
  posted_at?: string | null;
  salary?: string | null;
  extensions?: string[];
  detected_extensions?: Record<string, any>;
  apply_options?: JobApplyOption[];
  job_highlights?: JobHighlight[];
  share_link?: string | null;
};

export type JobSearchResponse = {
  query: string;
  location: string;
  page: number;
  employment_type?: string | null;
  role_filters?: string[];
  seniority_filters?: string[];
  jobs: JobListing[];
};

export type JobDetailResponse = {
  job: JobListing;
};

export type SavedJob = {
  job_id: string;
  saved_at: string;
  job: JobListing;
};

export type SavedJobsResponse = {
  jobs: SavedJob[];
};
